src/utils/internal/multipartParse.ts

```diff
@@ -0,0 +1,81 @@
+import { DefaultMultipartBodyType } from '../handlers/requestHandler'
+
+function parseContentHeaders(
+  headersString: string,
+): { name: string; filename?: string; contentType: string } {
+  const headers = new Headers(
+    headersString.split('\r\n').map((s) => {
+      const [name, ...rest] = s.split(': ')
+      return [name, rest.join(': ')]
+    }),
+  )
+  const contentType = headers.get('content-type') ?? 'text/plain'
+  const disposition = headers.get('content-disposition')
+  if (!disposition) {
+    throw new Error('"Content-Disposition" header is required.')
+  }
+  const directives: Record<string, string | undefined> = {}
+  disposition.split(';').map((s) => {
+    const [name, ...rest] = s.trim().split('=')
+    directives[name] = rest.join('=')
+  })
+  const { name: _name = '', filename: _filename } = directives
+  const name = _name.slice(1, -1)
+  const filename = _filename === undefined ? undefined : _filename.slice(1, -1)
+  return { name, filename, contentType }
+}
+
+/**
+ * Parses a given string as a multipart/form-data.
+ * Does not throw an exception on an invalid multipart string.
+ */
+export function multipartParse<T extends DefaultMultipartBodyType>(
```

Comment:
What do you think if we name this function `parseMultipartData`?

Reply:
A function `jsonParse` already exists. Should I also rename it?

Reply:
I would leave `jsonParse` as-is. You're right to aim for consistency here. I believe `jsonParse` was named to resemble `JSON.parse`. We can address it separately if you want.